test/e2e/common.go

```diff
@@ -501,3 +509,62 @@ func (t *Tester) PodLog(podName, nameSpace string, lines int64) (string, error)
 
 	return buf.String(), nil
 }
+
+func (t *Tester) CreateManagedClusterAddOn(managedClusterNamespace, addOnName string) error {
+	_, err := t.AddOnClinet.AddonV1alpha1().ManagedClusterAddOns(managedClusterNamespace).Create(
+		context.TODO(),
+		&addonv1alpha1.ManagedClusterAddOn{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace:   managedClusterNamespace,
+				Name:        addOnName,
+				Annotations: map[string]string{"addon.open-cluster-management.io/installNamespace": addOnName},
+			},
+		},
+		metav1.CreateOptions{},
+	)
+	return err
+}
+
+func (t *Tester) CreateManagedClusterAddOnLease(addOnInstallNamespace, addOnName string) error {
+	if _, err := t.KubeClient.CoreV1().Namespaces().Create(
+		context.TODO(),
+		&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: addOnInstallNamespace,
+			},
+		},
+		metav1.CreateOptions{},
+	); err != nil {
+		return err
+	}
+
+	_, err := t.KubeClient.CoordinationV1().Leases(addOnInstallNamespace).Create(
+		context.TODO(),
+		&coordv1.Lease{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      addOnName,
+				Namespace: addOnInstallNamespace,
+			},
+			Spec: coordv1.LeaseSpec{
+				RenewTime: &metav1.MicroTime{Time: time.Now()},
+			},
+		},
+		metav1.CreateOptions{},
+	)
+	return err
+}
+
+func (t *Tester) CheckManagedClusterAddOnStatus(managedClusterNamespace, addOnName string) error {
+	addOn, err := t.AddOnClinet.AddonV1alpha1().ManagedClusterAddOns(managedClusterNamespace).Get(context.TODO(), addOnName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	if addOn.Status.Conditions == nil {
+		return fmt.Errorf("there is no conditions in addon %v/%v", managedClusterNamespace, addOnName)
+	}
+	cond := meta.FindStatusCondition(addOn.Status.Conditions, "ManagedClusterAddOnConditionAvailable")
```

Comment:
nit: meta.IsStatusConditionTrue() will be simpler

Reply:
changed